### ISSUE TITLE
redis err: dial tcp 127.0.0.1:6379: getsockopt: connection refused

### ISSUE THREAD

**Issue by penhauer-xiao** [oliver006/redis_exporter#79]
Through the command redis-cli 127.0.0.1 -p zzzaa001 connection redis is good. But the redis_exporter error is as follows:

```
[root@acserver ~]# docker run -e "REDIS_ADDR=127.0.0.1:6379" -e "REDIS_PASSWORD=zzzaa001" --name redis_exporter -p 9121:9121 oliver006/redis_exporter
time="2017-06-21T09:12:59Z" level=info msg="Redis Metrics Exporter v0.11.1    build date: 2017-05-24-14:10:15    sha1: d3af2b49709a2cab654d7c876497e791c6d9a082    Go: go1.8.1\n" 
time="2017-06-21T09:12:59Z" level=info msg="Providing metrics at :9121/metrics" 
time="2017-06-21T09:12:59Z" level=info msg="Connecting to redis hosts: []string{\"127.0.0.1:6379\"}" 
time="2017-06-21T09:12:59Z" level=info msg="Using alias: []string{\"\"}" 
time="2017-06-21T09:13:03Z" level=info msg="redis err: dial tcp 127.0.0.1:6379: getsockopt: connection refused" 
time="2017-06-21T09:13:08Z" level=info msg="redis err: dial tcp 127.0.0.1:6379: getsockopt: connection refused" 
time="2017-06-21T09:13:13Z" level=info msg="redis err: dial tcp 127.0.0.1:6379: getsockopt: connection refused" 
```

And my  prometheus cofnig is:

```
  - job_name: redis_exporter_dev

    scrape_interval: 5s

    static_configs:
    - targets: ['192.168.1.2:9121']

  - job_name: redis_exporter_test

    scrape_interval: 5s

    static_configs:
    - targets: ['192.168.1.3:9121']

```
Use default redisaddr the error as below:

```
[root@acserver ~]# docker run -e "REDIS_PASSWORD=zzzaa001" --name redis_exporter -p 9121:9121 oliver006/redis_exporter
time="2017-06-21T09:32:22Z" level=info msg="Redis Metrics Exporter v0.11.1    build date: 2017-05-24-14:10:15    sha1: d3af2b49709a2cab654d7c876497e791c6d9a082    Go: go1.8.1\n" 
time="2017-06-21T09:32:22Z" level=info msg="Providing metrics at :9121/metrics" 
time="2017-06-21T09:32:22Z" level=info msg="Connecting to redis hosts: []string{\"redis://localhost:6379\"}" 
time="2017-06-21T09:32:22Z" level=info msg="Using alias: []string{\"\"}" 
time="2017-06-21T09:32:22Z" level=info msg="redis err: dial redis: unknown network redis"
```

**Comment by oliver006** [comment]
You'll need to provide the address of the container that runs your redis instance - 127.0.0.1:6379 I think is relative to the docker container so this won't work.

**Comment by penhauer-xiao** [comment]
@oliver006  Redis service is not docker way to deploy, but in this machine using docker way to start redis_export, I try to use 127.0.0.1:6379 or the machine's other addresses have failed.  as the same issue

**Comment by oliver006** [comment]
How do you run your redis server in the above example?

**Comment by penhauer-xiao** [comment]
@oliver006 
```
[root@~]# ps -ef | grep redis
redis      917     1  0 Jun18 ?        00:04:29 /usr/bin/redis-server 127.0.0.1:6379
```

**Comment by penhauer-xiao** [comment]
@oliver006
I put redis bind to 0.0.0.0 to start, the problem remains the same.
`[root@acserver ~]# ps -ef | grep redis
root     23500     1  0 13:53 ?        00:00:01 /usr/bin/redis-server 0.0.0.0:6379

[root@acserver ~]# docker run -e "REDIS_ADDR=127.0.0.1:6379" -e "REDIS_PASSWORD=zzzaa001" --name redis_exporter -p 9121:9121 oliver006/redis_exporter
time="2017-06-22T05:54:13Z" level=info msg="Redis Metrics Exporter v0.11.1    build date: 2017-05-24-14:10:15    sha1: d3af2b49709a2cab654d7c876497e791c6d9a082    Go: go1.8.1\n" 
time="2017-06-22T05:54:13Z" level=info msg="Providing metrics at :9121/metrics" 
time="2017-06-22T05:54:13Z" level=info msg="Connecting to redis hosts: []string{\"127.0.0.1:6379\"}" 
time="2017-06-22T05:54:13Z" level=info msg="Using alias: []string{\"\"}" 

time="2017-06-22T06:02:56Z" level=info msg="redis err: dial tcp 127.0.0.1:6379: getsockopt: connection refused" 

`

**Comment by penhauer-xiao** [comment]
This problem has been resolved, can not use 127.0.0.1 to connect.

**Comment by penhauer-xiao** [comment]
This problem has been resolved, can not use 127.0.0.1 to connect.
